playbooks/roles/caasp4/tasks/cleanup_nodes.yml

```diff
@@ -19,6 +19,9 @@
 # stack in setup_nodes.yml so we need to remove the resources again from
 # terraform state and let heat handle the reource cleanup
 
+- name: Assert required packages are alredy present
+  include: _assert_packages.yml
+
```

Comment:
Why do you need it during cleanup? You're not deleting packages/repos anyway...

Reply:
But the part you are linking is merely testing if a package is installed, on the other hand your code installs the packages even during the cleanup, that does not seem to make sense

Reply:
It will install it **only if** its not there. In existing code, if its not there, then it will just error out.
In this change, it will install (a corrective action instead of just throwing error) it if its missing . 
IMHO, this situation is highly unlikely as cleanup is needed only after initial setup. 
What do you recommend ?

Reply:
Well I just do not see a reason to install packages when you're doing cleanup action. Not that it is such big issue

Reply:
So let's assume if terraform is not there while doing cleanup, what user needs to do ? 

I am not aware of the real reason for having this check in cleanup and that's why following existing logic in this change. I am assuming its there so that terraform can be used for cleaning resources.

If preference is to not have this _assert_packages include, I can certainly remove it as it seems highly unlikely we will have this situation.
Let me know as I am not hung up on this idea.

Reply:
well, so if terraform runs and deploys caasp4 terrafon+skuba will alreaqdy be installed and they are used for the teardown. If it didnt run, then they wont be installed.

The outcome is the same, if teardown tries to cleanup caasp4 and there is no terraform binary it will fail. But in that case we should not care, should we? If there is no terraform binary it means caasp4 was not deployed, so we can safely skip the caasp teardown.

So it should be safe to drop this in the teardown and add an assert for the terraform binary in the caasp cleanup so we dont break the cleanup task on clean_caasp failure.